packages/react-router/lib/server-runtime/build.ts

```diff
@@ -24,14 +24,20 @@ export interface ServerBuild {
   };
   routes: ServerRouteManifest;
   assets: AssetsManifest;
-  basename?: string;
+  // `| undefined` is required to ensure compatibility with TypeScript's
+  // `exactOptionalPropertyTypes` option
+  basename?: string | undefined;
```

Comment:
```suggestion
  basename: string | undefined;
```

Reply:
Are you still experiencing issues without this change? If so, are you able to reproduce this issue in the test?

Reply:
Yes, so I should investigate why this test does not fail!